bundle/config/mutator/apply_presets.go

```diff
@@ -22,6 +22,8 @@ type applyPresets struct{}
 
 // Apply all presets, e.g. the prefix presets that
 // adds a prefix to all names of all resources.
+//
+// Note that the catalog/schema presets are applied in ApplyPresetsCatalogSchema.
```

Comment:
`apply_presets` is becoming a bit too large. For the new addition, I moved everything to a separate file. Lots of boring but easy to read/test code there.

As a followup we should look into a further refactoring: either factoring the logic by `preset` feature or by resource. For some of the overarching tests it's certainly been better to do it by feature, but we could end up with some combination for the tests.